### DISCUSSION TITLE
GitHub Package Registry - NuGet - 400 Bad Request

### DISCUSSION THREAD

**Discussion by AptiviCEO** [community/community#26181] [Packages]
<p>I am trying to upload a package for modders on my repository (console application) and I have set up nuget.exe using the GitHub Package Registry configuration for NuGet steps on Help.</p>
<p>When I tried to push my .nupkg file to GitHub, using nuget.exe, I got an error after uploading, saying 400 Bad Request.</p>
<p>I have set the API key to the source, because I think it is a root cause, but when I tried to upload it with new changes, the same 400 Bad Request error appears.</p>
<p>I used nuget push KS.0.0.6.13.symbols.nupkg -Source “GitHub - KS” for uploading to GitHub NuGet repo. I have set the site as “<a href="https://nuget.pkg.github.com/EoflaOE/index.json" rel="nofollow noopener">https://nuget.pkg.github.com/EoflaOE/index.json</a>” and username as “EoflaOE”.</p>
<p>My repository is .NET Framework 4.7.x console application, and is found at “<a href="https://github.com/EoflaOE/Kernel-Simulator" rel="nofollow noopener">https://github.com/EoflaOE/Kernel-Simulator</a>”, for future reference. It is a kernel simulator that can be an API for making modifications that compile and run inside an app.</p>

**Comment by warrenbuckley** [comment]
<p>Did you ever resole or figure this out <a class="mention" href="/eoflaoe">@eoflaoe</a> ?</p>
<p>As I have the same issue with my package by trying to use the new GitHub Actions CI YML Workflows</p>
<p><a href="https://github.com/warrenbuckley/Take-Out-The-Trash/runs/196791261" rel="nofollow noopener">https://github.com/warrenbuckley/Take-Out-The-Trash/runs/196791261</a></p>

**Comment by AptiviCEO** [comment]
<p>No <a class="mention" href="/warrenbuckley">@warrenbuckley</a>, nobody knew yet.</p>

**Comment by AptiviCEO** [comment]
<p><a class="mention" href="/warrenbuckley">@warrenbuckley</a> Did you get the notification saying that you’re accepted? I didn’t.</p>
<p>I contacted them and they said that I am on the list but not accepted yet.</p>

**Comment by warrenbuckley** [comment]
<p>Hmm I am pretty sure I got an email to say I was accepted for this.</p>
<p>I have a support ticket I am waiting to hear back from on how to use this in GitHub Actions, but they have not told me that I am NOT in the BETA, so I assume I am.</p>

**Comment by AptiviCEO** [comment]
<p>Still, I did not get an email. Can you please tell them to add me to the beta? If you do, I will let you be a member of my project.</p>

**Comment by AptiviCEO** [comment]
<p><a class="mention" href="/warrenbuckley">@warrenbuckley</a> See the above message</p>

**Comment by stoiveyp** [comment]
<p>Just raised a support ticket for similar NuGet problems - I get “invalid response” from GH Package Repo every time I try to push, but NuGet works fine so I know the package is okay. On package repo and actions beta so I’ll let you know if I hear anything</p>

**Comment by AptiviCEO** [comment]
<p><a class="mention" href="/stoiveyp">@stoiveyp</a> Thanks! The problem on my end is that GitHub didn’t accept my beta request, and I have to wait for months. What can I do to let them speed up?</p>

**Comment by tonyfonager** [comment]
<blockquote>
<p><a class="mention" href="/stoiveyp">@stoiveyp</a> wrote:</p>
</blockquote>
<blockquote>
<p>Just raised a support ticket for similar NuGet problems - I get “invalid response” from GH Package Repo every time I try to push, but NuGet works fine so I know the package is okay. On package repo and actions beta so I’ll let you know if I hear anything</p>
</blockquote>
<p>I have exactly the same problem - got granted access yesterday, but it fails whenever I try to push to the repository.</p>

**Comment by AptiviCEO** [comment, marked as answer]
<p>I have been accepted, pushed a package, and works fine. Did you put the GitHub username as the username, and the API key as a password? <a class="mention" href="/tonyfonager">@tonyfonager</a> </p>

**Reply by DennyWebb** [reply]
Our issue ended up being that i didn't have the namespace in the package name.  namspace/name vs just name.

**Comment by tonyfonager** [comment]
<p>I got it working, but it actually relies on the content of the NuGet package.</p>
<p>I tried making a new “Hello World” package, and I could deploy that without issues to GPR.</p>
<p>But all other projects fails.</p>
<p>And I should maybe mention that whenever they fail, I try nuget push’ing them to my Azure Artifacts feed, and that works ecvery time without issues.</p>

**Comment by AptiviCEO** [comment]
<p>Well, I can upload NuGet packages with debugging symbols to GitHub Package Registry successfully. Do you, by chance, have put in unusual variables and values in your nuspec? You can send them a ticket by emailing GitHub with the title of “GitHub Package Registry”. <a class="mention" href="/tonyfonager">@tonyfonager</a> </p>

**Comment by toanxyz** [comment]
<p>I have the same problem too. We are trying to push nuget packages to GitHub Package Registry via Azure DevOps pipeline.</p>

**Comment by AptiviCEO** [comment]
<p><a class="mention" href="/toanxyz">@toanxyz</a> Are you sure that you are authenticated correctly, and that you have used your API key as the password? You can e-mail GitHub.</p>

**Comment by sappenin** [comment]
<p>I’m getting this error as well. I was able to upload a single version last Friday, but yesterday and today I get error 400 when trying to upload artifacts.</p>

**Comment by rdubya16** [comment]
<p><a class="mention" href="/tonyfonager">@tonyfonager</a> Can you elaborate on this? What content of the nuget package matters? I am having the same issue and am unable to determine what is causing it.</p>

**Comment by vslee** [comment]
<p>For those who can’t get this to work, we have come up with a workaround using curl here <a href="https://github.com/NuGet/Home/issues/8580#issuecomment-555696372" rel="nofollow noopener">https://github.com/NuGet/Home/issues/8580#issuecomment-555696372</a>, at least until GitHub fixes problem of not allowing the apiToken to work directly using the nuget push command. </p>

**Comment by stoiveyp** [comment]
<p>Thank you! The team did work with me for a while and ended up in a place where the suggested fix was using a NuSpec file. But your workaround does the trick and means I can keep my other processes as needed</p>

**Comment by AptiviCEO** [comment]
<p>For anyone who haven’t specified RepositoryUrl and RepositoryType, you might get this error. Check <a href="https://github.community/t/github-package-registry-and-github-actions-500-bad-request/14666/2">this link</a> for more details.</p>
